src/IO/ReadBufferFromS3.cpp

```diff
@@ -84,44 +78,49 @@ bool ReadBufferFromS3::nextImpl()
     ProfileEvents::increment(ProfileEvents::S3ReadMicroseconds, watch.elapsedMicroseconds());
     if (!next_result)
         return false;
-    internal_buffer = impl->buffer();
+
+    working_buffer = internal_buffer = impl->buffer();
+    pos = working_buffer.begin();
 
     ProfileEvents::increment(ProfileEvents::S3ReadBytes, internal_buffer.size());
 
-    working_buffer = internal_buffer;
+    offset += working_buffer.size();
+
     return true;
 }
 
 off_t ReadBufferFromS3::seek(off_t offset_, int whence)
 {
-    if (impl)
-        throw Exception("Seek is allowed only before first read attempt from the buffer.", ErrorCodes::CANNOT_SEEK_THROUGH_FILE);
-
```

Comment:
If you allow to seek in `ReadBufferFromS3`, it is worth to remove that functionality from `ReadIndirectBufferFromRemoteFS`.

Reply:
Okay, it actually dead code, will return back old seek behavior.